### PR TITLE
Fixes exiting/crash handling/ctrl+c

### PR DESCRIPTION
**Pull request by kamronbatman**
- [X] Fixes handling of CTRL+C
- [X] Fixes crash guard (but should really be rewritten
- [X] Fixes exiting/restarting

TODO: Create a mode where restarting won't actually restart to support linux daemons.

### CODE REVIEW

File: Projects/UOContent/Misc/CrashGuard.cs [hunk 1]
```diff
@@ -8,32 +8,38 @@ namespace Server.Misc
...
-        private static readonly bool GenerateReport = true;
+        // TODO: Make this configurable
+        private const bool Enabled = true;
+        private const bool SaveBackup = true;
+        private const bool RestartServer = true;
+        private const bool GenerateReport = true;
 
         public static void Initialize()
         {
             if (Enabled) // If enabled, register our crash event handler
```
Comment (deccer) [inline Projects/UOContent/Misc/CrashGuard.cs:19]: Obvious comments should be removed

**Comment by deccer** [comment]
hmm really odd. im still in git bash to pull/compile modernuo, even enter uo path but then again:

```cs
$ ./ModernUO.exe
ModernUO - [https://github.com/modernuo/modernuo] Version 0.6.3.15
Core: Running on .NET Core 3.1.7

Core: Optimizing for 8 processors
Core: High resolution timing (Supported)
Core: Server configuration is missing data directories.
Please enter the absolute path to the Ultima Online data:
 > C:\Temp\_Temp\Ultima Online Classic
Core: Path C:\Temp\_Temp\Ultima Online Classic added.
[finish] >
Core: Server is missing socket listener IP addresses.
Please enter the IP and ports to listen:
 - Only enter IP addresses directly bound to this machine
 - To listen to all IP addresses enter 0.0.0.0
[0.0.0.0:2593]>
Core: Server configuration saved to Configuration/modernuo.json.
Warning: Server.Factions.FactionSawTrap
       - No serialization constructor

Core: Reading email configuration from Configuration/email-settings.json...done
Regions: Loading...done. (385 regions, 0 failures) (0.65 seconds)
World: Loading...done (0 items, 0 mobiles) (0.02 seconds)
ServerList: Auto-detecting public IP address...done (41.223.78.70)
This server has no accounts.
Do you want to create the owner account now? (y/n)Error:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.InvalidOperationException: Cannot read keys when either application does not have a console or when console input has been redirected. Try Console.Read.
   at System.ConsolePal.ReadKey(Boolean intercept)
   at System.Console.ReadKey(Boolean intercept)
   at Server.Misc.AccountPrompt.Initialize() in C:\Code\_Extern\ModernUO\Projects\UOContent\Misc\AccountPrompt.cs:line 15
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Server.AssemblyHandler.Invoke(String method) in C:\Code\_Extern\ModernUO\Projects\Server\AssemblyHandler.cs:line 56
   at Server.Core.Main(String[] args) in C:\Code\_Extern\ModernUO\Projects\Server\Main.cs:line 450
This exception is fatal, press return to exit
```

**Review by deccer — APPROVED** [review]
Besides the comment, I mentioned, which could be removed. this looks alright.

File: Projects/UOContent/Misc/CrashGuard.cs (hunk 1 above)
Comment (kamronbatman) [reply Projects/UOContent/Misc/CrashGuard.cs:19]: Yeah legacy stuff. Good call.

**Comment by deccer** [comment]
No crash so far. Looks like this is working.

**Pull request merged**